apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -194,12 +212,19 @@ def _opener(
         """Opens and returns a TileDB object specific to this type."""
         raise NotImplementedError()
 
-    def reopen(self, mode: options.OpenMode, timestamp: OpenTimestamp | None) -> clib.SOMAObject:
+    def reopen(self, mode: options.OpenMode, timestamp: OpenTimestamp | None) -> Wrapper[_RawHdl_co]:
+        """Returns a new copy of the wrapper handle in the requested mode and at the requested timestamp."""
         if mode not in ("r", "w"):
             raise ValueError(f"Invalid mode '{mode}' passed. " "Valid modes are 'r' and 'w'.")
         ts = self.context._open_timestamp_ms(timestamp)
         self.metadata._write()
-        return self._handle.reopen(clib.OpenMode.read if mode == "r" else clib.OpenMode.write, (0, ts))
+        clib_handle = self._handle.reopen(clib.OpenMode.read if mode == "r" else clib.OpenMode.write, (0, ts))
+        return self.__class__.open(
+            uri=clib_handle.uri,
+            mode=clib.mode,
+            context=self.context,
+            timestamp=clib.timestamp,
+        )
```

Comment:
We are re-opening twice here. This is not new, but it is more evident now that the code is more isolated.